v8.go

```diff
@@ -77,14 +77,19 @@ type Snapshot struct{ data C.StartupData }
 
 func newSnapshot(data C.StartupData) *Snapshot {
 	s := &Snapshot{data}
-	runtime.SetFinalizer(s, (*Snapshot).release)
```

Comment:
By wrapping, you mean creating a new struct that is just wrapping the context? If so, I like that. It would still be nice to expose release, while still leaving the finalizers. With release being exposed then the user can decide when to clean up, but if they forget then the finalizer will protect them.